### PR TITLE
feat(pagination): add helpers

### PR DESCRIPTION
**Pull request by hekike**
Add helpers to:

- Parse query params to `Page`
- Create `PagedResponse` from existing with transforming items type

### CODE REVIEW

File: pkg/pagination/pagination.go [hunk 1]
```diff
@@ -50,6 +75,25 @@ type PagedResponse[T any] struct {
 	Items      []T  `json:"items"`
 }
 
+// NewPagedResponse creates a new PagedResponse with the given page, totalCount and items.
+func NewPagedResponseWith[OUT any, IN any](resp PagedResponse[IN], m func(in IN) (OUT, error)) (PagedResponse[OUT], error) {
```
Comment (GAlexIHU) [inline pkg/pagination/pagination.go:79]: I'm not convinced we need this. e.g.

```go
apiRes := pagination.PagedResponse{
  Page: res.Page,
  TotalCount: res.TotalCount,
  Items: slicesx.Map(...)
}
```

looks fine to me...

File: pkg/pagination/pagination.go [hunk 2]
```diff
@@ -20,6 +20,31 @@ type Page struct {
...
+func NewPage(pageNumber int, pageSize int) Page {
+	return Page{
+		PageSize:   pageSize,
+		PageNumber: pageNumber,
+	}
+}
+
+// NewPageFromRef creates a new Page from pointers to pageNumber and pageSize.
+// Useful for handling query parameters.
+func NewPageFromRef(pageNumber *int, pageSize *int) Page {
```
Comment (GAlexIHU) [inline pkg/pagination/pagination.go:33]: IMO if you want this utility you should move it to `pkg/framework/commonhttp`, it's an API specific thing we do

Comment (hekike) [reply pkg/pagination/pagination.go:33]: I agree, but this is true for `PaginatedResponse` true + `NewPageFromRef` doesn't enforce is a query param, we would need to make an interface for it.

File: pkg/pagination/pagination.go (hunk 1 above)
Comment (hekike) [reply pkg/pagination/pagination.go]: The challenge is if that then we have to copy struct properties everywhere in router which is easy to forget + error prone if we add new properties to PagedResponse like hasNextPage

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**